main.tf

```diff
@@ -86,3 +88,47 @@ resource "aws_security_group_rule" "hcp_consul" {
   to_port           = local.hcp_consul_security_groups[count.index].port
   type              = "ingress"
 }
+
```

Comment:
I couldn't quite figure out a better way of merging the two cases 1) a user provides `var.security_group_ids` and 2) a user doesn't provide it.

One potential alternative is to have a new variable called `var.create_security_group` which will control these resources instead of using `length(var.security_group_ids) == 0`

Reply:
Good call, will add that!

Reply:
@crhino it might be beneficial to add a list of data sources for each of the var.security_group_ids and then use a for_each  ?